api-service/service/hosts.go

```diff
@@ -49,41 +53,66 @@ func (as *APIService) SearchHostsAsLMS(filters dto.SearchHostsFilters) (*exceliz
 		return nil, aerr
 	}
 
+	var j int
 	for i, val := range hosts {
 		i += 4 // offset for headers
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("B%d", i), val["physicalServerName"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("C%d", i), val["virtualServerName"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("D%d", i), val["virtualizationTechnology"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("E%d", i), val["dbInstanceName"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("F%d", i), val["pluggableDatabaseName"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("G%d", i), val["environment"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("H%d", i), val["options"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("I%d", i), val["usedManagementPacks"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("N%d", i), val["productVersion"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("O%d", i), val["productLicenseAllocated"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("P%d", i), val["licenseMetricAllocated"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("Q%d", i), val["usingLicenseCount"])
-
-		hostname := val["physicalServerName"].(string)
-		if len(hostname) == 0 {
-			hostname = val["virtualServerName"].(string)
-		}
-		if csi, ok := csiByHostname[hostname]; ok {
-			lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("R%d", i), strings.Join(csi, ", "))
+		setCellValueLMS(lms, sheetDatabaseEbsDbTier, i, csiByHostname, val)
+		createdDate := val["createdAt"].(primitive.DateTime).Time().UTC()
+		if (filters.From != utils.MIN_TIME ||
+			filters.To != utils.MAX_TIME) &&
+			createdDate.After(filters.From) &&
+			createdDate.Before(filters.To) {
+			if j == 0 {
+				j = 4 // offset for headers
+			}
+			if j == 4 {
+				indexsheetHostAdded := lms.NewSheet(sheetHostAdded)
+				iindexSheetDatabaseEbsDbTier := lms.GetSheetIndex(sheetDatabaseEbsDbTier)
+				errs := lms.CopySheet(iindexSheetDatabaseEbsDbTier, indexsheetHostAdded)
+				if errs != nil {
+					return nil, errs
+				}
+				lms.SetActiveSheet(iindexSheetDatabaseEbsDbTier)
```

Comment:
Is this needed?
Why?

Reply:
Yes, we need lms.SetActiveSheet(iindexSheetDatabaseEbsDbTier) because without this statement the active sheet is Validation that is a hidden sheet.

Reply:
Adding a new sheet change the active sheet!
Very nice, thank you!